After running `node-reinstall ` I get npm ERR! Error: ENOENT, open 'package.js

I tried this on my mac but get these errors. I originally installed node as sudo, maybe that is my issue..

![image](https://cloud.githubusercontent.com/assets/2746655/6967502/9dccd274-d957-11e4-8e0f-46b3eb6786b4.png)

```
npm ERR! addLocal Could not install .
npm ERR! Error: ENOENT, open 'package.json'
npm ERR! If you need help, you may report this *entire* log,
npm ERR! including the npm and node versions, at:
npm ERR!     <http://github.com/npm/npm/issues>

npm ERR! System Darwin 12.5.0
npm ERR! command "/Users/me/.nvm/v0.10.38/bin/node" "/Users/me/.nvm/v0.10.38/bin/npm" "install" "--global"
npm ERR! cwd /Users/me
npm ERR! node -v v0.10.38
npm ERR! npm -v 1.4.28
npm ERR! path package.json
npm ERR! code ENOENT
npm ERR! errno 34
```

Okay, I'm pretty sure this is the case. Steps to reproduce:

* Create an empty directory somewhere
* run this command `npm install --global`

I got this output which appears identical to what you got:
```
npm ERR! addLocal Could not install .
npm ERR! Error: ENOENT, open 'package.json'
npm ERR! If you need help, you may report this *entire* log,
npm ERR! including the npm and node versions, at:
npm ERR!     <http://github.com/npm/npm/issues>

npm ERR! System Darwin 14.1.0
npm ERR! command "/Users/brock/.nvm/v0.10.37/bin/node" "/Users/brock/.nvm/v0.10.37/bin/npm" "install" "--global"
npm ERR! cwd /Users/brock/research/test-reinstall
npm ERR! node -v v0.10.37
npm ERR! npm -v 1.4.28
npm ERR! path package.json
npm ERR! code ENOENT
npm ERR! errno 34
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/brock/research/test-reinstall/npm-debug.log
npm ERR! not ok code 0
```

I'll work on an update that confirms that there were global modules installed.

Great work!

On Thursday, 2 April 2015, Brock Angelo <notifications@github.com> wrote:

> Okay, I'm pretty sure this is the case. Steps to reproduce:
>
>    - Create an empty directory somewhere
>    - run this command npm install --global
>
> I got this output which appears identical to what you got:
>
> npm ERR! addLocal Could not install .
> npm ERR! Error: ENOENT, open 'package.json'
> npm ERR! If you need help, you may report this *entire* log,
> npm ERR! including the npm and node versions, at:
> npm ERR!     <http://github.com/npm/npm/issues>
>
> npm ERR! System Darwin 14.1.0
> npm ERR! command "/Users/brock/.nvm/v0.10.37/bin/node" "/Users/brock/.nvm/v0.10.37/bin/npm" "install" "--global"
> npm ERR! cwd /Users/brock/research/test-reinstall
> npm ERR! node -v v0.10.37
> npm ERR! npm -v 1.4.28
> npm ERR! path package.json
> npm ERR! code ENOENT
> npm ERR! errno 34
> npm ERR!
> npm ERR! Additional logging details can be found in:
> npm ERR!     /Users/brock/research/test-reinstall/npm-debug.log
> npm ERR! not ok code 0
>
> I'll work on an update that confirms that there were global modules
> installed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/brock/node-reinstall/issues/13#issuecomment-88987498>.
>


-- 

Kind regards
Nikos

--- a/node-reinstall.sh
+++ b/node-reinstall.sh
@@ -3,7 +3,7 @@
 # credit: http://stackoverflow.com/a/11178106/2083544
 
 ## program version
-VERSION="0.0.4"
+VERSION="0.0.5"
 
 ## path prefix
 PREFIX="${PREFIX:-/usr/local}"
@@ -171,7 +171,8 @@ elif (( $USE_NAVE )); then
   nave usemain $NODE_VERSION
 fi
 
-echo "Reinstalling your global npm modules:"
-echo $GLOBAL_MODULES
-
-npm install --global $GLOBAL_MODULES
+if [ ${#GLOBAL_MODULES[@]} -gt 0 ]; then
+  echo "Reinstalling your global npm modules:"
+  echo $GLOBAL_MODULES
+  npm install --global $GLOBAL_MODULES
+fi

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "node-reinstall",
-  "version": "0.0.4",
+  "version": "0.0.5",
   "description": "Completely re-install Node, NPM and NVM",
   "scripts": [ "node-reinstall.sh" ],
   "install": "make install",

